go/batch-submitter/txmgr/txmgr.go

```diff
@@ -229,14 +238,24 @@ func WaitMined(
 
 	for {
 		receipt, err := backend.TransactionReceipt(ctx, txHash)
-		if receipt != nil {
-			return receipt, nil
-		}
+		switch {
+		case receipt != nil:
+			txHeight := receipt.BlockNumber.Uint64()
+			tipHeight, err := backend.BlockNumber(ctx)
+			if err != nil {
+				log.Error("Unable to fetch block number", "err", err)
+				break
+			}
 
-		if err != nil {
+			if txHeight+numConfirmations <= tipHeight+1 {
```

Comment:
Could this be simplified to `txHeight+numConfirmations < tipHeight`? Just curious about the `+1`

Reply:
To maintain the same behavior I think it'd be the opposite, `txHeight+numConfirmations` would have to be strictly less than `tipHeight+2`.

Another way to write this check is `txHeight+numConfirmations-1 <= tipHeight`. The `-1` comes from the fact that confirmations have an inherent "off-by-one", i.e. plug `numConfirmations=1` into the equation and we would expect the transaction to be confirmed at `txHeight=tipHeight`. The equation is written in the form above to avoid underflows. I will add some comments